operator/pkg/gateway-api/cell.go

```diff
@@ -60,6 +60,32 @@ func registerController(lc hive.Lifecycle, config gatewayApiConfig, ctrlRuntimeM
 	return nil
 }
 
+// registerReconcilers registers the Gateway API reconcilers to the controller-runtime library manager.
+func registerReconcilers(mgr ctrlRuntime.Manager, enableSecretSync bool, secretsNamespace string, idleTimeoutSeconds int) error {
+	reconcilers := []interface {
+		SetupWithManager(mgr ctrlRuntime.Manager) error
+	}{
+		newGatewayClassReconciler(mgr),
+		newGatewayReconciler(mgr, secretsNamespace, idleTimeoutSeconds),
+		newReferenceGrantReconciler(mgr),
+		newHTTPRouteReconciler(mgr),
+		newGRPCRouteReconciler(mgr),
+		newTLSRouteReconciler(mgr),
+	}
+
+	if enableSecretSync {
+		reconcilers = append(reconcilers, newSecretSyncReconciler(mgr, secretsNamespace))
+	}
+
+	for _, r := range reconcilers {
+		if err := r.SetupWithManager(mgr); err != nil {
+			return fmt.Errorf("failed to setup reconciler: %w", err)
```

Comment:
(nit) similarly, here, can the reconciler have a human readable name so that the error says which reconciler failed to be setup?

Reply:
I thought about the same. The reconciler has a name, but unfortunately we're re-using the names between different reconcilers - so they aren't of that much help either.

Will follow up on this in a separate PR.